account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -26,3 +49,208 @@ export function alchemyGasManagerMiddleware(
     context: { policyId: policyId },
   });
 }
+
+interface AlchemyGasAndPaymasterAndDataMiddlewareParams {
+  policyId: string;
+  transport: AlchemyTransport;
+  enableDummyPaymasterAndData?: boolean;
+}
+
+/**
+ * Paymaster middleware factory that uses Alchemy's Gas Manager for sponsoring transactions using
+ * the `alchemy_requestGasAndPaymasterAndData` method instead of standard ERC-7677 methods.
+ *
+ * @example
+ *  ```ts
+ * import { sepolia, alchemyGasAndPaymasterAndDataMiddleware } from "@account-kit/infra";
+ * import { alchemy } from "@account-kit/infra";
+ *
+ * const client = createAlchemySmartAccountClient({
+ *  transport: alchemy({ apiKey: "your-api-key" }),
+ *  chain: sepolia,
+ *  ...alchemyGasAndPaymasterAndDataMiddleware({
+ *    policyId: "policyId",
+ *    transport: alchemy({ apiKey: "your-api-key" }),
+ *  })
+ * });
+ * ```
+ *
+ * @param {AlchemyGasAndPaymasterAndDataMiddlewareParams} params configuration params
+ * @param {AlchemyGasAndPaymasterAndDataMiddlewareParams.policyId} params.policyId the policyId for Alchemy's gas manager
+ * @param {AlchemyGasAndPaymasterAndDataMiddlewareParams.transport} params.transport fallback transport to use for fee estimation when not using the paymaster
+ * @param {AlchemyGasAndPaymasterAndDataMiddlewareParams.enableDummyPaymasterAndData} params.enableDummyPaymasterAndData whether to enable the dummy paymaster and data middleware, which typically is only needed if the client is using a custom gasEstimator or feeEstimator
+ * @returns {Pick<ClientMiddlewareConfig, "dummyPaymasterAndData" | "paymasterAndData">} partial client middleware configuration containing `dummyPaymasterAndData` and `paymasterAndData`
+ */
+export function alchemyGasAndPaymasterAndDataMiddleware(
+  params: AlchemyGasAndPaymasterAndDataMiddlewareParams
+): Pick<
+  ClientMiddlewareConfig,
+  "dummyPaymasterAndData" | "feeEstimator" | "gasEstimator" | "paymasterAndData"
+> {
+  const { policyId, transport, enableDummyPaymasterAndData } = params;
+  return {
+    dummyPaymasterAndData: async (uo, args) => {
+      if (!args.client.chain) {
+        throw new ChainNotFoundError();
+      }
+
+      if (!enableDummyPaymasterAndData) {
+        // TODO(jh): Is this safe to do? Is my thinking that we normally don't need to actually do anything in
+        // this middleware function if using the default feeEstimator/gasEstimator correct, b/c the paymasterAndData
+        // middleware function should handle everything?
+        return noopMiddleware(uo, args);
+      }
+
+      if (bypassPaymasterAndData(args.overrides)) {
+        // No reason to generate stub data if we are bypassing the paymaster.
+        return noopMiddleware(uo, args);
+      }
+
+      const paymaster = getAlchemyPaymasterAddress(args.client.chain); // throws if unsupported chain
```

Comment:
❗ ❗ ❗ this is problematic b/c this function only supports 15 chains & throws if it's an unsupported chain.

@andysim3d do you think we should fall back to using `pm_getPaymasterStubData` here so we don't have to maintain this list of chains & paymaster addresses?